src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobDispatcher.java

```diff
@@ -39,4 +51,44 @@ public void dispatch(
             targetStorageAccount
         );
     }
+
+    private void uploadContent(
+        BlockBlobClient sourceBlob,
+        String destinationContainer,
+        TargetStorageAccount targetStorageAccount
+    ) {
+        logger.info("Upload inner zip  from blob {} to Container: {}", sourceBlob.getBlobUrl(), destinationContainer);
+        long startTime = System.nanoTime();
+        try (var zipStream = new ZipInputStream(sourceBlob.openInputStream());) {
+            ZipEntry entry;
+
+            while ((entry = zipStream.getNextEntry()) != null) {
+                if (Objects.equals(entry.getName(), ENVELOPE)) {
+
+                    blobContainerClientProxy.runUpload(
+                        sourceBlob, destinationContainer,
+                        targetStorageAccount,
+                        target -> blobMover.uploadWithChunks(target, zipStream)
+                    );
+
+                    logger.info(
+                        "Streaming finished for  blob {} to Container: {}, Upload Duration: {} sec",
+                        sourceBlob.getBlobUrl(),
+                        destinationContainer,
+                        TimeUnit.NANOSECONDS.toSeconds(System.nanoTime() - startTime)
+                    );
+                    return;
+                }
+            }
```

Comment:
Move these lines to `uploadEnvelope()` method and `InvalidZipArchiveException` from that method. This will make code better readable.

Reply:
where is this method?

Reply:
It should be a new method.

Reply:
in that case that new method return the status so caller can throw InvalidZipArchiveException. I think it wouldn't make more readable

Reply:
I added new method but it  throws 'InvalidZipArchiveException'

Reply:
For me - it is easier to understand.